projects/storefrontapp-e2e-cypress/cypress/helpers/customer-ticketing/customer-ticketing.ts

```diff
@@ -0,0 +1,202 @@
+const FIRST_ROW = 1;
+const SUBJECT_COLUMN = 1;
+const CATEGORY_COLUMN = 2;
+const STATUS_COLUMN = 5;
+
+const CUSTOMER_SUPPORT_MENU_OPTION_INDEX = 14;
+const FIRST_TICKET_COLUMN_INDEX = 1;
+const FIRST_TICKET_ROW_INDEX = 1;
+
+export enum TestCategory {
+  enquiry = "Enquiry",
+  complaint = "Complaint",
+  problem = "Problem",
+}
+
+export interface TestTicketDetails {
+  subject: string;
+  message: string;
+  category: TestCategory;
+  associatedTo?: string;
+  filename?: string;
+}
+
+export function loginRegisteredUser() {
+  login();
+}
+
+export function clickMyAccountMenuOption(){
+  cy.visit('/');
+  cy.get('#cx-header', { timeout: 10000 }).should('be.visible');
+  cy.onMobile(() => {
+    clickHamburger();
+  });
+  cy.get('.accNavComponent button').click();
+}
+
+export function clickCustomerSupportMenuOption(){
+  cy.get(`.accNavComponent li:nth-child(${CUSTOMER_SUPPORT_MENU_OPTION_INDEX})`).should('contain.text', 'Customer Service');
+  cy.get(`.accNavComponent li:nth-child(${CUSTOMER_SUPPORT_MENU_OPTION_INDEX}) a`).click();
+}
+
+export function verifyTicketListingPageVisit(){
+  cy.url().should('include','/my-account/support-tickets');
+  cy.get('cx-customer-ticketing-list').should('exist');
+}
+
+export function clickFirstTicketFromTicketListing(){
+  cy.get(`#ticketing-list-table tbody tr:nth-child(${FIRST_TICKET_ROW_INDEX}) .cx-ticketing-list-data:nth-child(${FIRST_TICKET_COLUMN_INDEX}) a.cx-ticketing-list-value`).should('exist');
+  cy.get(`#ticketing-list-table tbody tr:nth-child(${FIRST_TICKET_ROW_INDEX}) .cx-ticketing-list-data:nth-child(${FIRST_TICKET_COLUMN_INDEX}) a.cx-ticketing-list-value`).click();
+}
+
+export function verifyTicketDetailsPageVisit(){
+  cy.url().should('match',/http:\/\/.+\/my\-account\/support\-ticket\/[0-9]+/);
+  cy.get('cx-customer-ticketing-messages').should('exist');
+}
+
+export function visitElectronicTicketListingPage() {
+  visitHomePage();
+  const supportPage = waitForPage('/my-account/support-tickets', 'support-tickets');
+  cy.visit('/my-account/support-tickets');
+  cy.wait(`@${supportPage}`).its('response.statusCode').should('eq', HTTP_STATUS_OK);
+}
+
+
+export function openCreateTicketPopup() {
+  cy.get('cx-customer-ticketing-list').should('exist');
+  cy.get('cx-customer-ticketing-create button').click();
+  cy.get('cx-customer-ticketing-create-dialog').should('exist');
+}
+
+export function fillTicketDetails(ticketDetails: TestTicketDetails){
+  const CATEGORY_SELECT = 0;
+
+  cy.get('cx-customer-ticketing-create-dialog').within(() => {
+    cy.get('.cx-customer-ticket-form-container').should('contain', 'Add New Request');
+    cy.get('textarea').first().type(ticketDetails.subject);
+    cy.contains('.cx-customer-ticket-label', 'Category').get('select').eq(CATEGORY_SELECT).select(ticketDetails.category);
+    cy.get('textarea').last().type(ticketDetails.message);
+    cy.get('cx-form-errors').should('not.be.visible');
+  });
+}
+
+export function addFile(filename: string){
```

Comment:
I don't think this function is used in ticket list, same goes for other functions

Reply:
I feel I'm reviewing this code twice because this functions are helpers for other e2es, we must merge the other ones before reviewing maybe

Reply:
+1